src/ast/passes/semantic_analyser.cpp

```diff
@@ -2273,6 +2273,9 @@ void SemanticAnalyser::visit(VariableAddr &var_addr)
     if (!found->type.IsNoneTy()) {
       var_addr.var_addr_type = CreatePointer(found->type, found->type.GetAS());
     }
+    // We can't know if the pointer to a scratch variable was passed
+    // to an external function for assignment so just mark it as assigned.
+    found->was_assigned = true;
```

Comment:
Just FYI, we will need to apply type inference in the other direction at some point. E.g.

```
foo(int *x);

begin {
    let $y;
    foo(&$y); // This implies that y is an `int`.
}
```